shared/js/moz_intl.js

```diff
@@ -196,11 +209,15 @@ global.mozIntl = {
       minimumIntegerDigits: 2
     });
 
-    return navigator.mozL10n.formatValue('timePattern_' + type).then(fmt => {
+    const maxUnitIdx = getDurationUnitIdx(resolvedOptions.maxUnit, 0);
+    const minUnitIdx = getDurationUnitIdx(resolvedOptions.minUnit,
+      durationFormatOrder.length - 1);
+
+    return navigator.mozL10n.formatValue('durationPattern').then(fmt => {
       return {
```

Comment:
Nit: If you're only returning the object from the callback, you could also do:

    return navigator.mozL10n.formatValue('durationPattern').then(fmt => ({
      resolvedOptions: …
    }));